utils/storage/storage.js

```diff
@@ -248,10 +252,45 @@ class Storage {
     );
   }
 
+  fetchAnonymousId(key) {
+    let anonId;
+    switch (key) {
+      case "segment":
+        if (Store.enabled) {
+          anonId = Store.get(anonymousIdKeyMap[key]);
+        }
+        if (!anonId) {
+          Cookie.set("rudder_cookies", true);
+
+          if (Cookie.get("rudder_cookies")) {
+            Cookie.remove("rudder_cookies");
+            anonId = Cookie.get(anonymousIdKeyMap[key]);
+          }
+        }
+        return anonId;
+
+      default:
+        return undefined;
+    }
+  }
+
   /**
    * get stored anonymous id
    */
-  getAnonymousId() {
+  getAnonymousId(anonymousIdOptions) {
+    if (
+      anonymousIdOptions &&
+      anonymousIdOptions.autoCapture &&
+      anonymousIdOptions.autoCapture.enabled &&
+      typeof anonymousIdOptions.autoCapture.source === "string"
```

Comment:
Use `get` method instead.

Reply:
Done

Reply:
The object structure is incorrectly used.

Reply:
If we use `get` method in case the customer set 

`anonymousIdOptions: "random string"`

`get(anonymousIdOptions, "autoCapture.enabled") `
and
`get(anonymousIdOptions, "autoCapture.source")
`
is returning  "random string".

Reply:
That's an instrumentation problem and anyway it leads to incorrect results.

Reply:
Can you elaborate on 

> The object structure is incorrectly used.
> it leads to incorrect results.

I have tested it properly and the results don’t seem to be incorrect.

Reply:
The comment was for the previous 'non-get' implementation.

Reply:
I was referring to this commit,
https://github.com/rudderlabs/rudder-sdk-js/pull/497/commits/e2b3ca105ed158e4ac0a1360971788d42c688abd

where `autoCapture` object is missed while accessing the options.

Reply:
so are we good with this approach?

Reply:
No. Let's use the `get` method to access multi-level fields.

Reply:
Addressed